### PR TITLE
update CNN 12/10

### PR DESCRIPTION
**Pull request by sukcheon22**
Build a CNN architecture with one CNN layer

### CODE REVIEW

File: models/regression/simple_regression.py [hunk 1]
```diff
@@ -24,3 +24,41 @@ def _build_network(self) -> nn.Module:
...
+    def __init__(self, input_dim: int):
+        """
+            input_dim: number of features of input.
+            out_channels: parameter, can be changed depended on 
+            different purposes: combining a fully connected layer, etc. 
+            kernel_size set to 1 because there exists a dimension of 
+            input equal to 1.
+        """
+        super().__init__()
+        self.cnn = nn.Conv1d(in_channels=input_dim, out_channels=1, kernel_size=1)
```
Comment (nguyenhongson1996) [inline models/regression/simple_regression.py:38]:
```suggestion
        self.cnn = nn.Conv1d(in_channels=input_dim, out_channels=1, kernel_size=1)
        
```

File: models/regression/simple_regression.py [hunk 2]
```diff
@@ -24,3 +24,41 @@ def _build_network(self) -> nn.Module:
...
         return {}
+
+class CNN(nn.Module):
+    def __init__(self, input_dim: int):
+        """
+            input_dim: number of features of input.
+            out_channels: parameter, can be changed depended on 
+            different purposes: combining a fully connected layer, etc. 
+            kernel_size set to 1 because there exists a dimension of 
+            input equal to 1.
```
Comment (nguyenhongson1996) [inline models/regression/simple_regression.py:35]: We should use reStructuredText as the format

File: models/regression/simple_regression.py [hunk 3]
```diff
@@ -24,3 +24,41 @@ def _build_network(self) -> nn.Module:
...
+        """
+        super().__init__()
+        self.cnn = nn.Conv1d(in_channels=input_dim, out_channels=1, kernel_size=1)
+    def forward(self, data: torch.Tensor) -> torch.Tensor:
+        """
+            Build CNN with PyTorch
```
Comment (nguyenhongson1996) [inline models/regression/simple_regression.py:41]:
```suggestion
            Build CNN using PyTorch.
```

File: models/regression/simple_regression.py [hunk 4]
```diff
@@ -24,3 +24,41 @@ def _build_network(self) -> nn.Module:
...
+            unsqueeze: Add a dimension to data so as to be suitable for 
+            Conv1d (batch_size, input_dim) -> (batch_size, 1, input_dim).
+            permute: change location of dimension so that input_dim turns 
+            to be second dimension.
+            view(data.size(0), -1): flatten to 2D.
```
Comment (nguyenhongson1996) [inline models/regression/simple_regression.py:46]: We should move this description to the code part and use # to comment.

File: models/regression/simple_regression.py [hunk 5]
```diff
@@ -24,3 +24,41 @@ def _build_network(self) -> nn.Module:
...
+        """
+        data = data.unsqueeze(1)
```
Comment (nguyenhongson1996) [inline models/regression/simple_regression.py:48]:
```suggestion
        # Add a dimension to data so as to be suitable for Conv1d layer (batch_size, input_dim) -> (batch_size, 1, input_dim).
        data = data.unsqueeze(1)   
```

Comment (nguyenhongson1996) [reply models/regression/simple_regression.py:48]: Ditto for bellow. And why we use 1 as the dimension size

**Review by nguyenhongson1996 — CHANGES_REQUESTED** [review]
Can you also add the evaluation result to the PR description?

**Review by sukcheon22 — COMMENTED** [review]
Early stopping after 121 epochs, Train Loss: 8.1190, Val Loss: 9.4742

**Review by nguyenhongson1996 — APPROVED** [review]
LGTM. There are some small comments need to be addresed.
